### PR TITLE
[BEAM-2564] add integration test for string functions

### PR DESCRIPTION
**Pull request by xumingming**
Add integration test for string functions and fixed a bug in `BeamSqlTrimExpression`.

### CODE REVIEW

**Comment by xumingming** [comment]
R: @XuMingmin @takidau

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/string/BeamSqlTrimExpression.java [hunk 1]
```diff
@@ -63,16 +63,16 @@ public BeamSqlTrimExpression(List<BeamSqlExpression> operands) {
       return BeamSqlPrimitive.of(SqlTypeName.VARCHAR,
           opValueEvaluated(0, inputRecord).toString().trim());
     } else {
-      String type = opValueEvaluated(0, inputRecord);
+      SqlTrimFunction.Flag type = opValueEvaluated(0, inputRecord);
```
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/string/BeamSqlTrimExpression.java]: Is this a change in Calcite 1.13.0?

Comment (xumingming) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/string/BeamSqlTrimExpression.java]: Don't think so, previously I assume we can treat the `LEADING`, `TRAILING` etc as a string, but actually they are not, they are flags.

File: dsls/sql/src/test/java/org/apache/beam/dsls/sql/integrationtest/BeamSqlStringFunctionsIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        + "TRIM(LEADING ' ' FROM ' hello ') as tr1,"
+        + "TRIM(TRAILING ' ' FROM ' hello ') as tr2,"
+        + "TRIM(BOTH ' ' FROM ' hello ') as tr3,"
+        + "OVERLAY('w3333333rce' PLACING 'resou' FROM 3) as ol,"
+        + "SUBSTRING('hello' FROM 2) as ss,"
+        + "SUBSTRING('hello' FROM 2 FOR 2) as ss1,"
+        + "INITCAP('hello world') as ss1"
+    ;
+
+    PCollection<BeamSqlRow> rows = BeamSqlCli.compilePipeline(sql, pipeline, sqlEnv);
```
Comment (XuMingmin) [inline dsls/sql/src/test/java/org/apache/beam/dsls/sql/integrationtest/BeamSqlStringFunctionsIntegrationTest.java]: Let's write the integration test based on DSL methods, which is the main goal for merge to master.

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/string/BeamSqlTrimExpression.java (hunk 1 above)
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/string/BeamSqlTrimExpression.java]: good job to find the issue, that's what we want to do in these tests

**Comment by XuMingmin** [comment]
@takidau  can you have a peak here? I suppose this will be the base for the series of built-in function tests in BEAM-2559

**Comment by xumingming** [comment]
Yeah, this is the base for all the builtin functions test. Other PRs kinda
depend on it.

File: dsls/sql/src/test/java/org/apache/beam/dsls/sql/integrationtest/BeamSqlStringFunctionsIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  @Test
+  public void testStringFunctions() throws Exception {
+    String sql = "SELECT "
+        + "'hello' || ' world' as concat,"
+        + "CHAR_LENGTH('hello') as cl,"
+        + "CHARACTER_LENGTH('hello') as cl1,"
+        + "UPPER('hello') as up,"
+        + "LOWER('HELLO') as lo,"
+        + "POSITION('world' IN 'helloworld') as po,"
+        + "POSITION('world' IN 'helloworld' FROM 1) as po1,"
```
Comment (takidau) [inline dsls/sql/src/test/java/org/apache/beam/dsls/sql/integrationtest/BeamSqlStringFunctionsIntegrationTest.java]: The FROM part tells you where to search from? Might be nice to exercise that a little more thoroughly if so, e.g., POSITION('world' IN 'worldhelloworld' FROM 5) == 11.

**Comment by xumingming** [comment]
Updated the test for POSITION

**Comment by takidau** [comment]
Merged, feel free to close.

**Pull request closed without merging**